src/broker/shard_parser.c

```diff
@@ -317,8 +316,15 @@ sp_make_int_sp_value_from_string (SP_VALUE * value_p, char *pos, int length)
 {
   int result = 0;
   char tmp = pos[length];
+  char *p;
 
+  p = strchr (pos, ';');
+  if (p)
+    {
+      *p = '\0';
```

Comment:
I think that you need to save p's value and recovery it, if p's value is semicolon, 
as like 
char tmp = pos[length];
...
pos[length] = tmp;

Reply:
Sorry, I missed it. Thank you.